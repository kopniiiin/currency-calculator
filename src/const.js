export const Currency = {
  EUR: `EUR`,
  USD: `USD`,
  JPY: `JPY`
};

export const currenciesToTranslations = {
  [Currency.EUR]: `ÐµÐ²Ñ€Ð¾`,
  [Currency.USD]: `Ð´Ð¾Ð»Ð»Ð°Ñ€`,
  [Currency.JPY]: `Ð¸ÐµÐ½Ð°`
};

export const currenciesToEmoji = {
  [Currency.EUR]: `ðŸ’¶`,
  [Currency.USD]: `ðŸ’µ`,
  [Currency.JPY]: `ðŸ’´`
};

export const RATE_PRECISION = 2;
