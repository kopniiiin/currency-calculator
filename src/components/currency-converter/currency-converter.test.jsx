import React from "react";
import {shallow} from "enzyme";

import {Currency} from "../../const.js";

import CurrencyConverter from "./currency-converter.jsx";

describe(`snapshot test: CurrencyConverter`, () => {
  it(`should render correctly`, () => expect(
    shallow(
      <CurrencyConverter
        originalCurrency={Currency.EUR}
        finalCurrency={Currency.EUR}
        originalCurrencyAmount={`4`}
        finalCurrencyAmount={4}
        onOriginalCurrencyChange={() => {}}
        onFinalCurrencyChange={() => {}}
        onOriginalCurrencyAmountChange={() => {}}
        onSubmit={() => {}}/>
    )
  ).toMatchSnapshot());
});

describe(`e2e test: CurrencyConverter`, () => {
  it(`should call onSubmit and preventDefault`, () => {
    const onSubmit = jest.fn();
    const preventDefault = jest.fn();

    shallow(
      <CurrencyConverter
        originalCurrency={Currency.EUR}
        finalCurrency={Currency.EUR}
        originalCurrencyAmount={`4`}
        finalCurrencyAmount={4}
        onOriginalCurrencyChange={() => {}}
        onFinalCurrencyChange={() => {}}
        onOriginalCurrencyAmountChange={() => {}}
        onSubmit={onSubmit}/>
    ).find(`.currency-converter`).simulate(`submit`, {preventDefault});

    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(preventDefault).toHaveBeenCalledTimes(1);
  });
});
