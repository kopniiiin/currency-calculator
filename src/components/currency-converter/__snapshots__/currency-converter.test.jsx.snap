// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: CurrencyConverter should render correctly 1`] = `
<form
  className=" currency-converter"
  onSubmit={[Function]}
>
  <CurrencySelect
    activeCurrency="EUR"
    className="currency-converter__currency-select"
    mode="original"
    onChange={[Function]}
  />
  <NumberInput
    className=""
    label="Количество евро"
    name="original-currency-amount"
    onChange={[Function]}
    required={true}
    value="4"
  />
  <SubmitButton
    className="currency-converter__submit-button"
    label="Конвертировать валюту"
  />
  <CurrencySelect
    activeCurrency="EUR"
    className="currency-converter__currency-select"
    mode="final"
    onChange={[Function]}
  />
  <span
    aria-label="Количество евро 4"
    className="currency-converter__final-currency-amount"
  >
    4.00
  </span>
</form>
`;
